src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederatedData.java

```diff
@@ -238,12 +233,18 @@ private static ChannelInitializer<SocketChannel> createChannel(InetSocketAddress
 			@Override
 			protected void initChannel(SocketChannel ch) throws Exception {
 				final ChannelPipeline cp = ch.pipeline();
+				final Optional<ImmutablePair<ChannelInboundHandlerAdapter, ChannelOutboundHandlerAdapter>> compressionStrategy = FederationUtils.compressionStrategy();
 				cp.addLast("NetworkTrafficCounter", new NetworkTrafficCounter(FederatedStatistics::logServerTraffic));
 				if(ssl)
 					cp.addLast(createSSLHandler(ch, address));
 				if(timeout > -1)
 					cp.addLast(new ReadTimeoutHandler(timeout));
-				cp.addLast(FederationUtils.decoder(), new FederatedRequestEncoder(), handler);
+				// cp.addLast(FederationUtils.decoder(), new FederatedRequestEncoder(), handler);
+				compressionStrategy.ifPresent(strategy -> cp.addLast(strategy.left));
+				cp.addLast(FederationUtils.decoder());
+				compressionStrategy.ifPresent(strategy -> cp.addLast(strategy.right));
+				cp.addLast(new FederatedRequestEncoder());
```

Comment:
Is it correct when it always adds the decoder even if no compression is specified? 
Should it not only conditionally ad the decoder?
Or is this what the if Present does?

Reply:
Exactly, the consumer to add the decoder / encoder will only be executed if the optional value is not empty (see [java docs](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-))